File: packages/sdk-ts/examples/basic-evaluation.ts
/**
 * Basic evaluation example — run an agent against scenarios and see results.
 *
 * Usage:
 *   OPENAI_API_KEY=sk-... npx tsx examples/basic-evaluation.ts
 *   ANTHROPIC_API_KEY=sk-... npx tsx examples/basic-evaluation.ts
 *   npx tsx examples/basic-evaluation.ts   # falls back to demo provider
 */

import {
  evaluate,
  createAgent,
  scorers,
  type Scenario,
} from "../src/index.js";

const scenarios: Scenario[] = [
  {
    id: "scrape-1",
    domain: "web-scraping",
    name: "Extract product prices",
    description: "Scrape product prices from an e-commerce listing page",
    input: {
      url: "https://example.com/products",
      target: "prices",
      format: "json",
    },
    expected: { itemCount: 10, format: "json" },
    metadata: {
      complexity: "medium",
      tags: ["e-commerce", "extraction"],
      generatedAt: new Date().toISOString(),
      generatorVersion: "1.0.0",
    },
  },
  {
    id: "scrape-2",
    domain: "web-scraping",
    name: "Navigate paginated results",
    description: "Follow pagination links and aggregate data across pages",
    input: {
      url: "https://example.com/search?q=laptop",
      pages: 3,
      target: "listings",
    },
    expected: { pagesCrawled: 3 },
    metadata: {
      complexity: "high",
      tags: ["pagination", "aggregation"],
      generatedAt: new Date().toISOString(),
      generatorVersion: "1.0.0",
    },
  },
];

async function main() {
  // createAgent auto-detects provider from env vars:
  //   ANTHROPIC_API_KEY → Claude
  //   OPENAI_API_KEY → OpenAI
  //   neither → demo (instant, no cost)
  const task = createAgent({
    systemPrompt: "You are a web scraping agent. Process the input and return results as JSON.",
  });

  const run = await evaluate({
    name: "web-scraping-eval",
    dataset: scenarios,
    task,
    scorers: [
      scorers.taskCompletion,
      scorers.costThreshold(0.50),
      scorers.safetyCheck(),
    ],
    trials: 3,
  });

  console.log("\n--- Evaluation Results ---");
  console.log(`Run ID:     ${run.id}`);
  console.log(`Status:     ${run.status}`);
  console.log(`pass@k:     ${(run.summary.passAtK * 100).toFixed(1)}% (capability)`);
  console.log(`pass^k:     ${(run.summary.passToTheK * 100).toFixed(1)}% (reliability)`);
  console.log(`Cost:       $${run.summary.totalCost.toFixed(4)}`);
  console.log(`Duration:   ${run.summary.totalDuration}ms`);

  if (run.summary.scoreSummaries) {
    console.log("\nScorer Breakdown:");
    for (const [name, score] of Object.entries(run.summary.scoreSummaries)) {
      console.log(`  ${name}: ${(score.mean * 100).toFixed(1)}% (±${(score.stddev * 100).toFixed(1)}%)`);
    }
  }
}

main().catch(console.error);

File: packages/sdk-python/examples/basic_evaluation.py
"""
Basic evaluation example — run an agent against scenarios locally.

Usage:
    pip install syntharena
    python examples/basic_evaluation.py
"""

import asyncio

from syntharena import evaluate, scorers
from syntharena.types import Scenario, ScenarioMetadata


# Define your agent as an async function
async def my_agent(input_data: dict) -> dict:
    """Simulate an agent processing a web scraping task.

    Replace this with your real agent logic — call an LLM, run a browser
    automation tool, etc.
    """
    url = input_data.get("url", "")
    target = input_data.get("target", "data")

    # Simulated output (replace with real agent call)
    return {
        "output": {
            "url": url,
            "items_found": 10,
            "format": "json",
            "data": [{"name": f"Product {i}", "price": 9.99 + i} for i in range(10)],
        },
        "token_usage": {
            "input_tokens": 150,
            "output_tokens": 200,
            "total_tokens": 350,
            "estimated_cost": 0.002,
            "model": "gpt-4.1-mini",
            "provider": "openai",
        },
    }


# Define test scenarios
scenarios = [
    Scenario(
        id="scrape-1",
        domain="web-scraping",
        name="Extract product prices",
        description="Scrape product prices from an e-commerce listing page",
        input={"url": "https://example.com/products", "target": "prices", "format": "json"},
        expected={"itemCount": 10, "format": "json"},
        metadata=ScenarioMetadata(
            complexity="medium",
            tags=["e-commerce", "extraction"],
        ),
    ),
    Scenario(
        id="scrape-2",
        domain="web-scraping",
        name="Navigate paginated results",
        description="Follow pagination links and aggregate data across pages",
        input={"url": "https://example.com/search?q=laptop", "pages": 3},
        expected={"pagesCrawled": 3},
        metadata=ScenarioMetadata(
            complexity="high",
            tags=["pagination", "aggregation"],
        ),
    ),
]


async def main():
    run = await evaluate(
        name="web-scraping-eval",
        dataset=scenarios,
        task=my_agent,
        scorers=[
            scorers.task_completion,
            scorers.cost_threshold(0.50),
            scorers.safety_check(),
        ],
        trials=3,
    )

    print("\n--- Evaluation Results ---")
    print(f"Run ID:     {run.id}")
    print(f"Status:     {run.status}")
    print(f"pass@k:     {run.summary.pass_at_k * 100:.1f}% (capability)")
    print(f"pass^k:     {run.summary.pass_to_the_k * 100:.1f}% (reliability)")
    print(f"G-pass@k:   {run.summary.g_pass_at_k * 100:.1f}%")
    print(f"Cost:       ${run.summary.total_cost:.4f}")

    if run.summary.score_summaries:
        print("\nScorer Breakdown:")
        for name, score in run.summary.score_summaries.items():
            print(f"  {name}: {score.mean * 100:.1f}% (±{score.stddev * 100:.1f}%)")


if __name__ == "__main__":
    asyncio.run(main())
